src/test/kotlin/com/karumi/maxibonkata/KarumiHQsTest.kt

```diff
@@ -0,0 +1,98 @@
+
+class KarumiHQsTest : ShouldSpec() {
+    init {
+        "KarumiHQs" {
+            should("start the day with 10 maxibons") {
+                val office = KarumiHQs()
+
+                office.maxibonsLeft shouldBe 10
+            }
+
+            should("always has more than two maxibons in the fridge") {
+                forAll(DeveloperGenerator(), { developer ->
+                    val office = KarumiHQs()
+
+                    office.openFridge(developer)
+
+                    office.maxibonsLeft > 2
+                })
+            }
+
+            should("buy 10 more maxibons if there are less than 3 in the fridge") {
+                forAll(HungryDeveloperGenerator(), { developer ->
+                    val office = KarumiHQs()
+                    val initialMaxibons = office.maxibonsLeft
+
+                    office.openFridge(developer)
+
+                    val expectedMaxibons = calculateMaxibonsLeft(initialMaxibons, developer)
+                    office.maxibonsLeft == expectedMaxibons
+                })
+            }
+
+            should("never request more maxibons to the team using the chat if there are more than 2 in the fridge") {
+                forAll(NotSoHungryDeveloperGenerator(), { developer ->
+                    val chat = MockChat()
+                    val office = KarumiHQs(chat)
+                    office.openFridge(developer)
+
+                    chat.messageSent == null
+                })
+            }
+
+            should("always has more than two maxibons in the fridge even if some karumies grab maxibons in group") {
+                forAll(Gen.list(DeveloperGenerator()), { developers ->
+                    val office = KarumiHQs()
+
+                    office.openFridge(developers)
+
+                    office.maxibonsLeft > 2
+                })
+            }
+
+            should("buy 10 more maxibons if there are less than 2 in the fridge when grabbing maxibons in group") {
+                forAll(Gen.list(DeveloperGenerator()), { developers ->
+                    val office = KarumiHQs()
+                    val initialMaxibons = office.maxibonsLeft
+
+                    office.openFridge(developers)
+
+                    val expectedMaxibons = calculateMaxibonsLeft(initialMaxibons, developers)
+                    office.maxibonsLeft == expectedMaxibons
+                })
+            }
+        }
+    }
+
+    private fun calculateMaxibonsLeft(initialMaxibons: Int, dev: Developer): Int =
+            calculateMaxibonsLeft(initialMaxibons, listOf(dev))
+
+    private fun calculateMaxibonsLeft(initialMaxibons: Int, developers: List<Developer>): Int {
+        var maxibonsLeft = initialMaxibons
+        developers.forEach { developer ->
+            maxibonsLeft = Math.max(0, maxibonsLeft - developer.maxibonsToGrab)
+            if (maxibonsLeft <= 2) {
+                maxibonsLeft += 10
+            }
+        }
+        return maxibonsLeft
+    }
```

Comment:
I know this method is a copy of the production code. Remember it's used to explain the training attendants the that sometime part of our software can't be tested using just property based and it's used as an example of when you don't have to use properties and use regular unit tests or tables.